fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/email/service/EmailReadPlatformServiceImpl.java

```diff
@@ -130,46 +130,41 @@ public EmailData retrieveOne(final Long resourceId) {
     @Override
     public Collection<EmailData> retrieveAllPending(final SearchParameters searchParameters) {
         final String sqlPlusLimit = (searchParameters.getLimit() > 0) ? " limit 0, " + searchParameters.getLimit() : "";
-        final String sql = "select " + this.emailRowMapper.schema() + " where emo.status_enum = "
-                + EmailMessageStatusType.PENDING.getValue() + sqlPlusLimit;
+        final String sql = "select " + this.emailRowMapper.schema() + " where emo.status_enum = " + sqlPlusLimit;
```

Comment:
Isn't there a '?' missing here? ie 'where emo.status_enum = ?'. Or am I missing something and ? gets appended elsewhere?

Also I don't think this is vulnerable for SQL injection as we are apending the value directly from the enum - but of course it makes sense to use parameters rather than string concatenation.

Reply:
My bad. Good find, to fix to `where emo.status_enum = ?`